crypto/vrf/vrf.go

```diff
@@ -60,15 +59,15 @@ func GenerateKey(rnd io.Reader) (pk *PublicKey, sk *PrivateKey, err error) {
 	pkP.ToBytes(&pkBytes)
 
 	copy(sk[32:], pkBytes[:])
-	copy(pk[:], pkBytes[:])
-	return pk, sk, err
+	return
 }
 
-func (sk *PrivateKey) Public() []byte {
-	return ed25519.PrivateKey(sk[:]).Public().(ed25519.PublicKey)
+func (sk *PrivateKey) Public() (PublicKey, bool) {
+	pk, ok := ed25519.PrivateKey(*sk).Public().(ed25519.PublicKey)
+	return PublicKey(pk), ok
 }
 
-func (sk *PrivateKey) expandSecret() (x, skhr *[32]byte) {
+func (sk PrivateKey) expandSecret() (x, skhr *[32]byte) {
```

Comment:
Because the Private/PublicKeys are now slices of bytes which (even if passed by value/used as value receiver) already behave more like a pointer (which makes the pointer receiver obsolete, only the "header" gets copied). Also, it made it easier to treat it as a slice of byte (otherwise we would need to add a `*` everywhere where we do a byte operation). But I can change it back to pointer receivers.